### PR TITLE
Enter pain prompt

### PR DESCRIPTION
**Pull request by raissacavalcanti**
Added EnterPainPrompt.kt, which is a Card component that gets displayed in the Home Screen depending on if the user has entered their muscle Pain or Not

### CODE REVIEW

File: app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt:27]: There is already an existing card widget for this called `WidgetCard`. Could you please use this? This makes all cards in our app look more consistent. e.g. your card here does not have round corners. `WidgetCard` is already used in the code. There you can see, how to use it.

In order to have an orange border color, you need to add an additional parameter to `WidgetCard` that toggles whether it should have an orange border or not.

File: app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt:29]: If this is not needed anymore, please delete this line 😄

File: app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt:47]: 👍 nice

File: app/src/main/java/com/example/mytrainingpal/screens/Home.kt
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/screens/Home.kt:32]: For this I would suggest the following:
In the app's preferences (the app's personal key-value-store) I would save the date of the day the user entered sore muscles the last time. If the date is not today, we know that the prompt to enter the sore muscles should be shown.

Later (not part of this pr)
The person developing the sore muscle input needs to update this "last time entered sore muscles" date so the prompt disappears once the user enters their sore muscles.

**Review by MichaelBuessemeyer — CHANGES_REQUESTED** [review]
Hey @raissacavalcanti,

Thanks for this pr and sorry for my late reply. I was busy finishing my model pr. 

Please have a look at my comments and incorporate the changes mentioned. I hope it shouldn't be much :)

File: app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt
Comment (raissacavalcanti) [reply app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt]: Thanks Micha! :)
WidgetCard() now accepts a boolean parameter that defines the border color, therefore I had to add arguments to all composables that use WidgetCard().

EnterPainPrompt() is now a WidgetCard().

File: app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt
Comment (raissacavalcanti) [reply app/src/main/java/com/example/mytrainingpal/components/EnterPainPrompt.kt]: done.

File: app/src/main/java/com/example/mytrainingpal/screens/Home.kt
Comment (raissacavalcanti) [reply app/src/main/java/com/example/mytrainingpal/screens/Home.kt:32]: Should the app's preferences be declared and instantiated in the MainActivity file? 🤔

Comment (MichaelBuessemeyer) [reply app/src/main/java/com/example/mytrainingpal/screens/Home.kt:32]: Honestly: I have no idea. I haven't looked up, how the preferences work. I guess this link will be helpful: https://developer.android.com/develop/ui/views/components/settings/use-saved-values

File: app/src/main/java/com/example/mytrainingpal/components/MusclePainWidget.kt
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/components/MusclePainWidget.kt:35]: Why did you comment these lines out? I know it will be changed in the future, but I would keep these code lines active (in my opinion).

File: app/src/main/java/com/example/mytrainingpal/components/WidgetCard.kt
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/components/WidgetCard.kt:26]: Awesome 🎉

**Review by MichaelBuessemeyer — CHANGES_REQUESTED** [review]
Hej,

thanks for adapting my requested changes this fast. The onliest thing missing now is saving the last entered sore muscle data to the preferences. I linked the android docs tutorial for this. I hope this is not too complicated. If it is, just notify me and I'll try to help 😄

File: app/src/main/java/com/example/mytrainingpal/screens/Home.kt
Comment (MichaelBuessemeyer) [reply app/src/main/java/com/example/mytrainingpal/screens/Home.kt:32]: => As it is better to use the room model classes to find out whether there is an entry for today, this pr is blocked by #21 and needs to wait until #21 is merged.

**Comment by MichaelBuessemeyer** [comment]
blocked by #21

File: app/src/main/java/com/example/mytrainingpal/states/TodaysMusclePainEntryState.kt
Comment (MichaelBuessemeyer) [inline app/src/main/java/com/example/mytrainingpal/states/TodaysMusclePainEntryState.kt:26]: @raissacavalcanti here is a composable that only serves as a state. It uses the ViewModel to access all the musclepainentries and once they are loaded it tries to find today's muscle pain entry and return it.
All that was left to do for me was to go through all entries and find today's entry. You already listened to `allMusclePainEntries` correctly.

You can see how simple it is to use in the `Home.kt`.

**Review by MichaelBuessemeyer — APPROVED** [review]
Hey @raissacavalcanti,

I just finished getting the musclepainentry of today and rendering the prompt in case there is none. You can see my comment pointing out what I did.


Additionally, I did some code clean up. The pr looks good now so I'll merge it.

**Pull request merged**
